core/reference_data.py

```diff
@@ -59,3 +61,22 @@ def seed_geospatial_dim_table():
                 db.session.add(geospatial_record)
 
     db.session.commit()
+
+
+def seed_fund_table():
+    """
+    Input seed data to fund_dim table using the mappings in /tests/resources/fund_dim.csv.
+    """
+
+    resources = Path(__file__).parent / ".." / "tests" / "resources"
+    fund_df = pd.read_csv(resources / "fund_dim.csv")
+
+    try:
+        fund_df.to_sql("fund_dim", con=db.session.connection(), index=False, index_label="id", if_exists="append")
```

Comment:
What I've written is a bit of an "anti-pattern" in that there could be an ```IntegrityError``` that we would want to be raised. I'll look at a rewrite. I think the idea is we'll just do a direct SQL insertion every time we onboard a new fund - at least we haven't thought about it further than that - so I was envisioning this function being only used for testing.

Reply:
Yeah mine was written with a bit of ambiguity, not knowing if we'd ever have to reseed, but agree it's likely if a new fund is onboarded or new/different geospatial data is added we'd do it as a SQL insertion in a migration rather than using any internal function or script, so if these functions are only used for testing they can be a lot simpler. Interested to see where you go with a rewrite if you do one - if I update my function to be closer to yours I could remove how it's called from `load_example_data`.

Reply:
I updated mine to be closer to yours...we could write something like:
```
except IntegrityError as e:
    db.session.rollback()
    error_message = str(e.orig)

    if isinstance(e.orig, postgresql.exc.UniqueViolation):
        current_app.logger.info(
            "UniqueKey violation occurred when inserting data into the fund_dim table."
            "Please remove any duplicate rows before calling this helper function."
        )
    else:
        current_app.logger.error(f"IntegrityError occurred: {error_message}")
    db.session.close()
```
But it is a bit of a hacky anti-pattern. If what you've written works we can stick with that, and then it's easier to update the reference tables.

Reply:
Sounds good - I think the only difference between our functions now is that I had `geospatial_dim_df["id"] = None` after creating the `geospatial_dim_df` so that a 'real' DB wouldn't use the hardcoded UUIDs in the csv, but as we discussed if these will only be used for tests and any DB updates will be done via migrations than that's not an issue and I can remove that line from mine.